framework/include/utils/RankTwoTensorImplementation.h

```diff
@@ -1125,26 +1144,19 @@ template <typename T>
 RankTwoTensorTempl<T>
 RankTwoTensorTempl<T>::genRandomTensor(T scale, T offset)
 {
-  RankTwoTensorTempl<T> tensor;
-
+  RankTwoTensorTempl<T> result;
   for (auto i : make_range(N))
     for (auto j : make_range(N))
-      tensor(i, j) = (MooseRandom::rand() + offset) * scale;
-
-  return tensor;
+      result(i, j) = (MooseRandom::rand() + offset) * scale;
+  return result;
 }
 
 template <typename T>
 RankTwoTensorTempl<T>
 RankTwoTensorTempl<T>::genRandomSymmTensor(T scale, T offset)
 {
-  RankTwoTensorTempl<T> tensor;
-
-  for (auto i : make_range(N))
-    for (unsigned int j = i; j < N; j++)
-      tensor(i, j) = tensor(j, i) = (MooseRandom::rand() + offset) * scale;
-
-  return tensor;
+  auto r = [&]() { return (MooseRandom::rand() + offset) * scale; };
```

Comment:
`const`

Reply:
Same question as above. This doesn't look right to me.

Reply:
yep see above

Reply:
By the way the `const` on a lambda would not mean anything.